dali/pipeline/data/tensor_list.h

```diff
@@ -103,13 +112,28 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
     }
     this->SetLayout(layout);
 
-    for (size_t i = 0; i < other.size(); ++i) {
-      type.template Copy<SrcBackend, Backend>(
-          raw_mutable_tensor(i),
-          other[i].raw_data(),
-          other[i].size(), stream);
-      this->meta_[i].SetSourceInfo(other[i].GetSourceInfo());
-      this->meta_[i].SetSkipSample(other[i].ShouldSkipSample());
+    if (use_copy_kernel && (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned())) {
```

Comment:
Is it only GPU->GPU or pinned ->GPU? Do we want GPU->pinned as well?
I would add a check regarding the destination backed as well.